### PR TITLE
Small doc updates

### PR DESCRIPTION
**Pull request by dlrobertson**
- Update documentation about current support in the wire module
 - Ensure the possible panic is documented for `Ipv6Option::data_mut`

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -49,24 +49,35 @@
...
 //! The packet layer exists to provide a more structured way to work with packets than
 //! treating them as sequences of octets. It makes no judgement as to content of the packets,
 //! except where necessary to provide safe access to fields, and strives to implement every
 //! feature ever defined, to ensure that, when the representation layer is unable to make sense
 //! of a packet, it is still logged correctly and in full.
 //!
+//! ## Packet and representation layer support
+//! The packet and representation layer currently support the following packet representations:
+//!
+//!  - Ethernet
```
Comment (whitequark) [inline src/lib.rs]: There's no EthernetRepr.

Comment (dlrobertson) [reply src/lib.rs]: Ah, good catch added a note here.

Comment (dlrobertson) [reply src/lib.rs]: Would a table be better here? Something like the following:

| Protocol | Packet | Representation |
|----------|--------|----------------|
| Ethernet | Yes    | No             |
| ARP      | Yes    | Yes            |
| IPv4     | Yes    | Yes            |
| ICMPv4   | Yes    | Yes            |
| IPv6     | Yes    | Yes            |
| ICMPv6   | Yes    | Yes            |
| TCP      | Yes    | Yes            |
| UDP      | Yes    | Yes            |

**Comment by whitequark** [comment]
Implementing EthernetRepr would be better :)

**Comment by dlrobertson** [comment]
> Implementing EthernetRepr would be better :)

Ah, didn't know if there was a reason behind the missing implementation. Did my best to take a crack at implementing it.

**Comment by dlrobertson** [comment]
Weird... I guess I didn't push my changes that included the EthernetRepr

**Pull request merged**